compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaNoneTypeSymbol.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.ballerina.compiler.api.impl.symbols;
+
+import io.ballerina.compiler.api.ModuleID;
+import io.ballerina.compiler.api.symbols.NoneTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BNoType;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+/**
+ * Represents the absence of a type descriptor.
+ */
+public class BallerinaNoneTypeSymbol extends AbstractTypeSymbol implements NoneTypeSymbol {
+
+    public BallerinaNoneTypeSymbol(CompilerContext context, ModuleID moduleID, BNoType bNoType) {
+        super(context, TypeDescKind.NONE, bNoType);
+    }
+
+    @Override
+    public String signature() {
+        return null;
```

Comment:
Shall we return an empty string instead of returning `null`?

Reply:
Updated in e44f7d5745e607bd8a098a5063f2fc39ad909daa